src/libraries/Microsoft.Extensions.Logging.Abstractions/ref/Microsoft.Extensions.Logging.Abstractions.cs

```diff
@@ -155,6 +155,23 @@ public enum LogLevel
         Critical = 5,
         None = 6,
     }
+    public abstract class BufferedLogRecord
```

Comment:
@geeknoid 

A couple comments...

* Should we add `ActivityTraceFlags` on here as well? OTel spec requires TraceId, SpanId, & TraceFlags for correlation: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#log-and-event-record-definition

* Suggest making attributes nullable...

  ```diff
  -  IReadOnlyList<KeyValuePair<string, object?>> Attributes
  +  IReadOnlyList<KeyValuePair<string, object?>>? Attributes
  ```

  Just thinking some logs won't have any attributes so the model should match.

* The `abstract` and `virtual` stuff feels a bit odd. For really a DTO could we do...

```csharp
    public sealed class BufferedLogRecord
    {
        public required System.DateTimeOffset Timestamp { get; init; }
        public required Microsoft.Extensions.Logging.LogLevel LogLevel { get; init; }
        public required Microsoft.Extensions.Logging.EventId EventId { get; init; }
        public string? Exception { get; init; }
        public System.Diagnostics.ActivitySpanId? ActivitySpanId { get; init; }
        public System.Diagnostics.ActivityTraceId? ActivityTraceId { get; init;  }
        public int? ManagedThreadId { get; init; }
        public string? FormattedMessage { get; init; }
        public string? MessageTemplate { get; init; }
        public System.Collections.Generic.IReadOnlyList<System.Collections.Generic.KeyValuePair<string, object?>> Attributes { get; init; }
    }
```

Reply:
@noahfalk Should we add trace flags? Just another prop, not a big deal...

I thought it's fine to have the attributes non-nullable, it would just point to an empty attribute list if there is nothing there.

As for the virtual/abstract duality, that was a result of refactoring and not paying attention. This is now all abstract. This means whomever implements this abstraction gets to decide how to manage the underlying storage of each property.

Reply:
I'm happy to add trace flags if Blanch believes they would be useful.

Reply:
> The abstract and virtual stuff feels a bit odd. For really a DTO could we do...

The design goal behind this type currently is that it isn't an intermediate DTO, it is likely the type that was used for backing storage. If you guys want to change this into a DTO and copy all the property values in from some private implementation type that does the storage I'm fine with that too.